STEP/TypeVisitor.cs

```diff
@@ -418,13 +417,13 @@ public void Visit(AssNode n) { // Add constant check, needs to be in ST (i think
     }
 
     public void Visit(FuncStmtNode n) {
-        var symbol = _symbolTable.RetrieveSymbol(n.Id.Id) as FunctionSymTableEntry;
-        if (symbol is null) { // Should this be here?
-            throw new NoNullAllowedException("The retrieved symbol was not a function symbol table entry");
+        if (_symbolTable.RetrieveSymbol(n.Id.Id) is not FunctionSymTableEntry symbol) {
+            throw new TypeException(n, "The retrieved symbol was not a function symbol table entry");
```

Comment:
Lige en heads-up: Fejlbeskeden kan være misvisende hvis symbolet slet ikke er skrevet ind i symbol table. Så f.eks. hvis du kalder en funktion der ikke findes, så siger den også at "The retrieved symbol was not a function symbol table entry", hvor den måske i stedet burde sige at den ikke findes i et af de aktive scopes

Reply:
har forbedret det